iosdevapp/GenerateImageViewModel.swift

```diff
@@ -55,12 +56,20 @@ class GenerateImageViewModel: ObservableObject {
             print(error)
         }
     }
+    func saveImageToGallery() {
+        guard let imageURL = imageURL else {
+            return
+        }
+        DispatchQueue.global(qos: .background).async {
+            let data = try! Data(contentsOf: imageURL)
+            let image = UIImage(data: data)!
+            UIImageWriteToSavedPhotosAlbum(image, nil, nil, nil)
+        }
+    }
 }
```

Comment:
Suggestion: I would move this saving image functionality out of the view model to a tiny service and test it separately. You could mock this service when you test the view model. So you are actually not saving a real image when you run the view model tests.